src/main/java/com/minecolonies/coremod/client/gui/WindowAssignCitizen.java

```diff
@@ -0,0 +1,199 @@
+package com.minecolonies.coremod.client.gui;
+
+import com.minecolonies.api.util.LanguageHandler;
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.blockout.Pane;
+import com.minecolonies.blockout.controls.Button;
+import com.minecolonies.blockout.controls.ButtonHandler;
+import com.minecolonies.blockout.controls.Label;
+import com.minecolonies.blockout.views.ScrollingList;
+import com.minecolonies.blockout.views.Window;
+import com.minecolonies.coremod.MineColonies;
+import com.minecolonies.coremod.colony.CitizenDataView;
+import com.minecolonies.coremod.colony.ColonyView;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.colony.buildings.BuildingHome;
+import com.minecolonies.coremod.network.messages.AssignUnassignMessage;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentTranslation;
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.minecolonies.api.util.constant.TranslationConstants.*;
+
+/**
+ * Window for the hiring or firing of a worker.
+ */
+public class WindowAssignCitizen extends Window implements ButtonHandler
+{
+    /**
+     * Id of the done button in the GUI.
+     */
+    private static final String BUTTON_DONE = "done";
+
+    /**
+     * Id of the cancel button in the GUI.
+     */
+    private static final String BUTTON_CANCEL = "cancel";
+
+    /**
+     * Id of the citizen name in the GUI.
+     */
+    private static final String CITIZEN_LABEL = "citizen";
+
+    /**
+     * Id of the id label in the GUI.
+     */
+    private static final String ID_LABEL = "id";
+
+    /**
+     * Id of the citizen list in the GUI.
+     */
+    private static final String CITIZEN_LIST = "unemployed";
+
+    /**
+     * Link to the xml file of the window.
+     */
+    private static final String BUILDING_NAME_RESOURCE_SUFFIX = ":gui/windowAssignCitizen.xml";
+
+    /**
+     * Position of the id label of each citizen in the list.
+     */
+    private static final int CITIZEN_ID_LABEL_POSITION = 3;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "CITIZEN_ID_LABEL_POSITION" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)